tests/net_dynamic_hb.rs

```diff
@@ -169,7 +169,38 @@ fn do_drop_and_readd(cfg: TestConfig) {
     // Run the network:
     loop {
         let (node_id, step) = net.crank_expect();
-
+        if !net[node_id].is_faulty() {
+            // Verify that only contributions from expected participants can be present in the batch.
+            for batch in &step.output {
+                let expected_participants: Vec<_> = if awaiting_removal.contains(&node_id) {
+                    // The node hasn't removed the pivot node yet.
+                    pub_keys_add.keys()
+                } else if awaiting_addition.contains(&node_id) {
+                    // The node has removed the pivot node but hasn't added it back yet.
+                    pub_keys_rm.keys()
+                } else {
+                    // The node has added the pivot node back.
+                    pub_keys_add.keys()
+                }.collect();
+                // FIXME: Update the validators in VirtualNet when a validator is removed or added.
+                let correct_validator_num = expected_participants.len() * 2 / 3;
```

Comment:
…so it should probably be `expected_participants.len() * 2 / 3 + 1`.
Or `batch.contributions().count() * 3 > correct_validator_num * 2` below.